lib/blocktree/blocktree.go

```diff
@@ -111,15 +112,15 @@ func (bt *BlockTree) AddBlock(header *types.Header, arrivalTime time.Time) (err
 
 // GetAllBlocksAtNumber will return all blocks hashes with the number of the given hash plus one.
 // To find all blocks at a number matching a certain block, pass in that block's parent hash
-func (bt *BlockTree) GetAllBlocksAtNumber(hash common.Hash) (hashes []common.Hash) {
+func (bt *BlockTree) GetAllBlocksAtNumber(parent common.Hash) (hashes []common.Hash) {
```

Comment:
We can keep `hash` instead of `parent` since blocks with the same number may not have the same parent, ex:

```
A -> B -> C
 \-> D -> E

C and E have the same block number but different parents
```

```suggestion
func (bt *BlockTree) GetAllBlocksAtNumber(hash common.Hash) (hashes []common.Hash) {
```

Reply:
reverted